packages/protocol/src/action-protocol/edge-modification.ts

```diff
@@ -14,41 +14,98 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { isArray, isString } from '../utils/typeguard-util';
-import { isActionKind, Operation } from './base-protocol';
-import { ElementAndRoutingPoints } from './types';
+import { hasArrayProp, hasStringProp } from '../utils/type-util';
+import { Operation } from './base-protocol';
+import { Args, ElementAndRoutingPoints } from './types';
 
 /**
  * If the source and/or target element of an edge should be adapted, the client can send a `ReconnectEdgeOperation` to the server.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `ReconnectEdgeOperations`.
  */
-export class ReconnectEdgeOperation implements Operation {
-    static readonly KIND = 'reconnectEdge';
-    constructor(
-        public readonly edgeElementId: string,
-        public readonly sourceElementId: string,
-        public readonly targetElementId: string,
-        public readonly kind: string = ReconnectEdgeOperation.KIND
-    ) {}
+export interface ReconnectEdgeOperation extends Operation {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof ReconnectEdgeOperation.KIND;
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.